plot_customizations.py

this top level plotting call is generating a plotting error several layers down in the plotting utilities and the external library. The stack dump looks like this for the error in examples/tutorials/[plot_customizations.py](https://github.com/SCECcode/pycsep/blob/master/examples/tutorials/plot_customizations.py) file.

<pre>
/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/io/__init__.py:241: DownloadWarning: Downloading: https://naturalearth.s3.amazonaws.com/10m_physical/ne_10m_coastline.zip
  warnings.warn(f'Downloading: {url}', DownloadWarning)
generating gallery for tutorials... [ 29%] catalog_forecast_evaluation.py
generating gallery for tutorials... [ 43%] gridded_forecast_evaluation.py
/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/io/__init__.py:241: DownloadWarning: Downloading: https://naturalearth.s3.amazonaws.com/50m_cultural/ne_50m_admin_0_boundary_lines_land.zip
  warnings.warn(f'Downloading: {url}', DownloadWarning)
WARNING: /home/runner/work/pycsep/pycsep/examples/tutorials/plot_customizations.py failed to execute correctly: Traceback (most recent call last):
  File "/home/runner/work/pycsep/pycsep/examples/tutorials/plot_customizations.py", line 122, in <module>
    ax = plots.plot_spatial_dataset(numpy.log10(rate_sum), forecast.region,
  File "/home/runner/work/pycsep/pycsep/csep/utils/plots.py", line 996, in plot_spatial_dataset
    im = ax.pcolor(lons, lats, gridded, cmap=cmap, alpha=alpha, snap=True,
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 315, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 2012, in pcolor
    limits = result.get_datalim(self.transData)
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/matplotlib/collections.py", line 267, in get_datalim
    paths = [transform.transform_path_non_affine(p) for p in paths]
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/matplotlib/collections.py", line 267, in <listcomp>
    paths = [transform.transform_path_non_affine(p) for p in paths]
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/matplotlib/transforms.py", line 2426, in transform_path_non_affine
    return self._a.transform_path_non_affine(path)
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/mpl/geoaxes.py", line 186, in transform_path_non_affine
    proj_geom = self.target_projection.project_geometry(
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/crs.py", line 817, in project_geometry
    return getattr(self, method_name)(geometry, src_crs)
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/crs.py", line 972, in _project_polygon
    return self._rings_to_multi_polygon(rings, is_ccw)
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/cartopy/crs.py", line 1233, in _rings_to_multi_polygon
    multi_poly = sgeom.MultiPolygon(polygon_bits)
  File "/home/runner/micromamba/envs/csep-dev/lib/python3.9/site-packages/shapely/geometry/multipolygon.py", line 79, in __new__
    shell = ob[0]
TypeError: 'GeometryCollection' object is not subscriptable


Extension error (sphinx_gallery.gen_gallery):
Handler <function generate_gallery_rst at 0x7f877553ce50> for event 'builder-inited' threw an exception (exception: 'GeometryCollection' object is not subscriptable)
make: *** [Makefile:27: html] Error 2
</pre>

--- a/examples/tutorials/plot_customizations.py
+++ b/examples/tutorials/plot_customizations.py
@@ -107,7 +107,7 @@ rate_sum = forecast.region.get_cartesian(rate_sum)
 # We define the arguments and a global projection, centered at $lon=-180$
 
 plot_args = {'figsize': (10,6), 'coastline':True, 'feature_color':'black',
-             'projection': cartopy.crs.Robinson(central_longitude=-179.0),
+             'projection': cartopy.crs.Robinson(central_longitude=180.0),
              'title': forecast.name, 'grid_labels': False,
              'cmap': 'magma',
              'clabel': r'$\log_{10}\lambda\left(M_w \in [{%.2f},\,{%.2f}]\right)$ per '
@@ -119,9 +119,9 @@ plot_args = {'figsize': (10,6), 'coastline':True, 'feature_color':'black',
 # To plot a global forecast, we must assign the option ``set_global=True``, which is required by :ref:cartopy to handle
 # internally the extent of the plot
 
-# ax = plots.plot_spatial_dataset(numpy.log10(rate_sum), forecast.region,
-#                                show=True, set_global=True,
-#                                plot_args=plot_args)
+ax = plots.plot_spatial_dataset(numpy.log10(rate_sum), forecast.region,
+                               show=True, set_global=True,
+                               plot_args=plot_args)
 
 ####################################################################################################################################
 

